src/Chunk.cpp

```diff
@@ -1887,31 +1885,24 @@ bool cChunk::AddClient(cClientHandle * a_Client)
 
 void cChunk::RemoveClient(cClientHandle * a_Client)
 {
-	for (cClientHandleList::iterator itrC = m_LoadedByClient.begin(); itrC != m_LoadedByClient.end(); ++itrC)
+	std::remove(m_LoadedByClient.begin(), m_LoadedByClient.end(), a_Client);
```

Comment:
Right, if this isn't misleading naming, then nothing is. C++ standard authors should be ashamed.